tests/unit/actions/settings/test_settings.py

```diff
@@ -0,0 +1,151 @@
+from ansible_navigator.actions.settings import filter_content_keys
+from ansible_navigator.ui_framework import Color
+from ansible_navigator.ui_framework import CursesLinePart
+from ansible_navigator.ui_framework import Decoration
+
+
+@dataclass
+class ColorMenuTestEntry:
+    """A test for menu coloring."""
+
+    color: int
+    """The color for menu entry"""
+    comment: str
+    """Describe the test"""
+    decoration: int
+    """The test decoration"""
+    is_default: bool
+    """Is the current value equal to the default"""
+
+    def __str__(self):
+        """Provide a string representation.
+
+        :returns: The string representation of self
+        """
+        return self.comment
+
+
+ColorMenuTestEntries = (
+    ColorMenuTestEntry(
+        comment="default/green",
+        color=Color.GREEN,
+        decoration=Decoration.NORMAL,
+        is_default=True,
+    ),
+    ColorMenuTestEntry(
+        comment="not default/yellow",
+        color=Color.YELLOW,
+        decoration=Decoration.NORMAL,
+        is_default=False,
+    ),
+)
+
+
+@pytest.mark.parametrize(argnames="data", argvalues=ColorMenuTestEntries, ids=str)
+def test_color_menu(data: ColorMenuTestEntry):
+    """Test color menu for a val set to the default.
+
+    :param data: A test entry
+    """
+    entry = {"is_default": data.is_default}
+    assert color_menu(0, "", entry) == (data.color, data.decoration)
+
+
+@dataclass
+class ContentHeadingEntry:
+    """A test for content headings."""
+
+    color: int
+    """The color for menu entry"""
+    comment: str
+    """Describes the test"""
+    current_value: str
+    """The current value"""
+    default: str
+    """The default value"""
+    is_default: Optional[bool] = None
+    """Indicate if the current is equal to the default."""
+    name: str = "test settings entry"
+    """The name of the settings entry"""
+
+    def __post_init__(self):
+        """Set is_default after initialization."""
+        self.is_default = self.current_value == self.default
+
+    @property
+    def heading(self):
+        """Create the expected heading for this content.
+
+        :returns: The expected heading
+        """
+        if self.is_default:
+            heading = CONTENT_HEADING_DEFAULT
+        else:
+            heading = CONTENT_HEADING_NOT_DEFAULT
+        return heading.format(**asdict(self))
+
+    def __str__(self):
+        """Provide a string representation.
+
+        :returns: The string representation of self
+        """
+        return self.comment
+
+
+ContentHeadingEntries = (
+    ContentHeadingEntry(
+        color=Color.GREEN,
+        comment="same",
+        current_value="navigator",
+        default="navigator",
+    ),
+    ContentHeadingEntry(
+        color=Color.YELLOW,
+        comment="different",
+        current_value="ansible",
+        default="navigator",
+    ),
+)
+
+
+@pytest.mark.parametrize(argnames="data", argvalues=ContentHeadingEntries, ids=str)
```

Comment:
Using kwargs for the first two arguments is very weird here — I didn't even know it's allowed and never knew their internal names. It's common to use positional arguments syntax for these two:
```suggestion
@pytest.mark.parametrize("data", ContentHeadingEntries, ids=str)
```

Reply:
This seems to be an accurate use of keyword arguments: https://github.com/pytest-dev/pytest/blob/18c0cfc0de1793ebb6ed2f4606cb8236611a5f64/src/_pytest/mark/structures.py#L128

Yes it may be uncommon, I don't mind it all.  

It helps explain what these are, and provide types information.

![image](https://user-images.githubusercontent.com/18386516/155570435-071c7f79-ba52-43f0-891f-d8c6133bcd61.png)

![image](https://user-images.githubusercontent.com/18386516/155570572-eebb95f2-ba4e-4d55-be7c-969c85477b8b.png)

I prefer this and will probably follow this pattern moving forward.

Reply:
I don't think it's good to go against the commonly understood practices, though. Have you checked if `flake8-pytest` is able to lint kwargs the same way as posargs?